packages/contracts/src/PoolMetadataRegistry.sol

```diff
@@ -22,18 +23,48 @@ contract PoolMetadataRegistry is IPoolMetadataRegistry {
         _vault = vault;
     }
 
-    /// @notice Reverts unless the pool with the given ID is registered on Balancer.
-    /// @param poolId The ID of the pool to check registration status for.
-    modifier onlyRegisteredPool(bytes32 poolId) {
-        require(_isPoolRegistered(poolId), "Pool not registered");
+    /// @notice Wraps a single pool ID in an array.
+    /// @param poolId The pool ID to wrap.
+    function _wrapPoolId(bytes32 poolId) internal pure returns (bytes32[] memory) {
+        bytes32[] memory wrappedPoolId = new bytes32[](1);
+        wrappedPoolId[0] = poolId;
+        return wrappedPoolId;
+    }
+
+    /// @notice Reverts unless the pools with the given ID are registered on Balancer.
+    /// @param poolIds The IDs of the pool to check registration status for.
+    modifier onlyRegisteredPools(bytes32[] memory poolIds) {
+        for (uint256 i = 0; i < poolIds.length; i++) {
+            require(_isPoolRegistered(poolIds[i]), "Pool not registered");
+        }
         _;
     }
 
-    /// @notice Reverts unless the caller is allowed to call the function.
-    /// @param poolId The ID of the pool that the caller wants to access.
-    modifier authenticate(bytes32 poolId) {
+    /// @notice Reverts unless the caller is allowed to call update for every pool.
+    /// @param poolIds The IDs of the pools that the caller wants to access.
+    modifier authenticatePools(bytes32[] memory poolIds) {
         bytes32 actionId = getActionId(msg.sig);
-        require(_canPerform(actionId, poolId, msg.sender), "sender not allowed");
+        for (uint256 i = 0; i < poolIds.length; i++) {
+            bytes32 poolId = poolIds[i];
+            require(_canPerform(actionId, poolId, msg.sender), "sender not allowed");
+        }
+        _;
+    }
+
+    /**
+     * @notice Validate the size of two arrays to avoid exceeding the gas limit and ensure equal length.
```

Comment:
I think we could decide on one docstring pattern to use 
https://docs.soliditylang.org/en/v0.8.17/natspec-format.html?highlight=documentation#documentation-example